src/parser/core/Entity.ts

```diff
@@ -111,13 +111,27 @@ class Entity {
   getBuffStacks(
     spellId: number,
     forTimestamp: number | null = null,
-    bufferTime = 0,
-    minimalActiveTime = 0,
+    bufferTime: number = 0,
+    minimalActiveTime: number = 0,
     sourceID: number | null = null,
-  ) {
-    return (
-      this.getBuff(spellId, forTimestamp, bufferTime, minimalActiveTime, sourceID)?.stacks || 0
+  ): number {
+    const buff: TrackedBuffEvent | undefined = this.getBuff(
+      spellId,
+      forTimestamp,
+      bufferTime,
+      minimalActiveTime,
+      sourceID,
     );
+    const currentTimestamp = forTimestamp !== null ? forTimestamp : this.owner.currentTimestamp;
+
+    let maxStackForTimestamp: StackHistoryElement = { timestamp: 0, stacks: 0 };
+    buff?.stackHistory.forEach((stack) => {
+      if (maxStackForTimestamp.stacks < stack.timestamp && stack.timestamp < currentTimestamp) {
```

Comment:
looks like a typo here. this should be:

```suggestion
      if (maxStackForTimestamp.timestamp < stack.timestamp && stack.timestamp < currentTimestamp) {
```

Reply:
ups